ocrd_models/ocrd_models/ocrd_exif.py

```diff
@@ -36,29 +39,27 @@ def __init__(self, img):
         #      print(img.info)
         for prop in ['compression', 'photometric_interpretation']:
             setattr(self, prop, img.info[prop] if prop in img.info else None)
-        if img.format in ('TIFF', 'PNG') and 'dpi' in img.info:
-            self.xResolution = int(img.info['dpi'][0])
-            self.yResolution = int(img.info['dpi'][1])
-            if img.format == 'TIFF':
-                self.resolutionUnit = 'cm' if img.tag.get(296) == 3 else 'inches'
-            else:
-                self.resolutionUnit = 'inches'
-        elif img.format == 'JPEG' and 'jfif_density' in img.info:
-            self.xResolution = img.info['jfif_density'][0]
-            self.yResolution = img.info['jfif_density'][1]
-            self.resolutionUnit = 'cm' if img.info['jfif_unit'] == 2 else 'inches'
-        elif img.format == 'PNG' and 'aspect' in img.info:
-            self.xResolution = img.info['aspect'][0]
-            self.yResolution = img.info['aspect'][1]
+        if img.format == 'JPEG2000':
+            # XXX find a way to get pixel density in jpeg2000
+            self.xResolution = self.yResolution = 1
             self.resolutionUnit = 'inches'
         else:
-            #  if img.format == 'JPEG2000':
-            #      import sys
-            #      print('JPEG 2000 not supported yet :(', file=sys.stderr)
-            self.xResolution = 1
-            self.yResolution = 1
-            self.resolutionUnit = 'inches'
-        #  print('format=%s type=%s' % (img.format, type(self.xResolution))
+            if img.filename:
+                ret = run(['identify', '-format', r'%x %y %U', img.filename], check=False, stderr=PIPE, stdout=PIPE)
+            else:
+                with BytesIO() as bio:
+                    img.save(bio, format=img.format)
+                    ret = run(['identify', '-format', r'%x %y %U', '/dev/stdin'], check=False, stderr=PIPE, stdout=PIPE, input=bio.getvalue())
+            if ret.returncode:
+                if not which('identify'):
+                    raise Exception("The 'identify' command is not available. Install with 'sudo apt install imagemagick'")
+                raise ValueError("'identify' returned non-zero exit code %s. stderr: %s" % (ret.returncode, ret.stderr))
```

Comment:
Maybe catch `no decode delegate for this image format ...` specially? And why re-raise here, and not just fallback to `resolution = 1`, as before?

Reply:
> Maybe catch `no decode delegate for this image format ...` specially?

:+1:

> And why re-raise here, and not just fallback to `resolution = 1`, as before?

Mostly because I wanted to see what went wrong if something went wrong with the call. But logging the error with a fallback is the more sensible thing to do, agreed.